src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -297,6 +347,112 @@ export default {
     hasCta() {
       return !!this.$scopedSlots.cta;
     },
+    onArrowKey(key) {
+      const currentItemIndex = this.filteredItems.indexOf(this.currentActiveItem);
+      let nextItem;
+
+      if (currentItemIndex >= 0) {
+        if (key === ARROW_DOWN) {
+          const selectableItems = this.filteredItems.slice(currentItemIndex + 1).filter(item => !item.disabled);
+          if (selectableItems.length) {
+            [nextItem] = selectableItems;
+          }
+        } else {
+          const selectableItems = this.filteredItems.slice(0, currentItemIndex).filter(item => !item.disabled);
+          if (selectableItems.length) {
+            nextItem = selectableItems[selectableItems.length - 1];
+          }
+        }
+      } else {
+        const selectableItems = this.filteredItems.filter(item => !item.disabled);
+        if (selectableItems.length) {
+          [nextItem] = selectableItems;
+        }
+      }
+      this.activateItemViaKeyboardNavigation(nextItem);
+    },
+    onArrowUpKeyDown() {
+      if (!this.isMultiple || this.isOpen) {
+        this.onArrowKey(ARROW_UP);
+      }
+    },
+    onArrowDownKeyDown() {
+      if (!this.isMultiple || this.isOpen) {
+        this.onArrowKey(ARROW_DOWN);
+      }
+    },
+    onEnterOrSpaceKeyDown() {
+      if (this.isOpen) {
+        if (!this.isMultiple || !this.currentActiveItem) {
+          this.closeViaKeyboardNavigation();
+        } else {
+          const keyboardNavigation = true;
+          this.select(this.currentActiveItem, keyboardNavigation);
+        }
+      } else {
+        this.show();
+      }
+    },
+    onSearchFieldEnterKeyDown() {
+      this.closeViaKeyboardNavigation();
+    },
+    onTabKeyDown() {
+      this.closeViaKeyboardNavigation();
```

Comment:
I think we should be able to close the dropdown on ESC too. That's the default behaviour of `<select>`.